packages/react-components/src/components/utils/unit-controller-history-handler.ts

```diff
@@ -0,0 +1,84 @@
+import { TimelineChart } from 'timeline-chart/lib/time-graph-model';
+import { TimeGraphUnitController } from 'timeline-chart/lib/time-graph-unit-controller';
+
+export interface HistoryItem {
+    selectionRange: TimelineChart.TimeGraphRange | undefined;
+    viewRange: TimelineChart.TimeGraphRange;
+}
+
+export class UnitControllerHistoryHandler {
+    protected history: Array<HistoryItem>;
+    protected index = 0;
+    protected maxAllowedIndex = 0;
+    protected timeout?: ReturnType<typeof setTimeout>;
+    protected unitController: TimeGraphUnitController;
+    protected restoring = false;
+
+    constructor(uC: TimeGraphUnitController) {
+        this.history = [];
+        this.unitController = uC;
+    }
+
+    public addCurrentState(): void {
+        const { selectionRange, viewRange } = this.unitController;
```

Comment:
In some cases (e.g. making time selection by clicking) the user can make the same selection many times. Each of these create their own history entry. Maybe it shouldn't be allowed to have (consecutive) duplicates in the history?

Not sure if it's unit controller's fault to trigger a selection/view range change event when it hasn't really changed, though...